docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,552 @@
+|-----------------------|---------|----------|--------------------------------------------------------------------------------------------|
+| `Color`               | `Color` | Yes      | The color of the Visual.                                                                   |
+| `RenderIfTransparent` | `bool`  | No       | Specifies whether to render in case of the `MixColor` being transparent (default `false`). |
 
-**Table: Color visual properties**
 
-| Property   | Type    | Required | Description         |
-|----------|-------|--------|-------------------|
-| `MixColor` |  `Vector4` | Yes      |  The color to be used. |
 
 **Usage:**
+The following example illustrates how to use a ColorVisual visual map:
 
-The following example illustrates how to use a property map for a color visual. The visual is created with the factory `CreateVisual()` method.
 
-```
-private const int PROPERTY_REGISTRATION_START_INDEX = 10001000;
-private const int ColorVisualPropertyIndex = PROPERTY_REGISTRATION_START_INDEX + 1;
-private const int PrimitiveVisualPropertyIndex = PROPERTY_REGISTRATION_START_INDEX + 2;
+   ```
+   ColorVisual _colorVisual_1 = new ColorVisual();
 
-private VisualBase _colorVisual;
+   // obligatory properties
+   _colorVisual_1.Color = Color.Red;
 
-PropertyMap colorVisual = new PropertyMap();
-colorVisual.Add(Visual.Property.Type, new PropertyValue((int)Visual.Type.Color))
-           .Add(ColorVisualProperty.MixColor, new PropertyValue(_color));
-_colorVisual = VisualFactory.Get().CreateVisual(colorVisual);
+   // optional properties:
+   _colorVisual_1.Size = new Size2D(400,400);
+   _colorVisual_1.DepthIndex = 5;
 
-RegisterVisual(ColorVisualPropertyIndex, _colorVisual);
+   _visualView.AddVisual("RedRectangle", _colorVisual_1);
 
-/// Set the color visual depth index
-_colorVisual.DepthIndex = ColorVisualPropertyIndex;
-```
 
-<a name="gradientvisual"></a>
-## Gradient Visual
+   ColorVisual _colorVisual_2 = new ColorVisual();
+
+   // obligatory properties
+   _colorVisual_2.Color = new Color(0.0f, 0.0f, 1.0f, 0.8f);
+
+   // optional properties:
+   _colorVisual_2.Size = new Size2D(300,300);
+   _colorVisual_2.Origin = Visual.AlignType.BottomEnd;
+   _colorVisual_2.AnchorPoint = Visual.AlignType.BottomEnd;
+   // shifting the visual left and up 
+   _colorVisual_2.RelativePosition = new RelativeVector2(-0.1f, -0.1f);
+   // for values > 0 corners are rounded
+   _colorVisual_2.CornerRadius = 50.0f;
+   // setting a value lower than for the previous object causes it to be covered
+   // without changing this value, the objects are drawn in the order they were added
+   _colorVisual_2.DepthIndex = _colorVisual_1.DepthIndex - 1;
+
+   _visualView.AddVisual("BlueRectangle", _colorVisual_2);
+   ```
+
+
+**Figure: ColorVisual obtained from the above code (_visualView background color set to gray).**
```

Comment:
```suggestion 
 **Figure: ColorVisual after implementing example.**

Reply:
I would like to stay with these parentheses, ok?